static/src/javascripts/lib/user-timing.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import performanceAPI from 'lib/window-performance';
+
+const timings = {};
+const startDate = new Date().getTime();
+
+const getCurrentTime = (): number => {
+    if ('now' in performanceAPI) {
+        return performanceAPI.now();
+    }
+
+    return new Date().getTime() - startDate;
+};
+
+const mark = (label: string): void => {
+    if ('mark' in performanceAPI) {
+        performanceAPI.mark(label);
+    } else {
+        timings[label] = getCurrentTime();
+    }
+};
+
+// Returns the ms time when the mark was made.
+const getTiming = (label: string): ?number => {
+    if ('getEntriesByName' in performanceAPI) {
+        const perfMark = performanceAPI.getEntriesByName(label, 'mark')[0];
+
+        if (perfMark && 'startTime' in perfMark) {
+            return perfMark.startTime;
+        }
+    } else if (label in timings) {
+        return timings[label];
+    }
+
+    return undefined;
```

Comment:
Okay. I question the usefulness of `consistent-return`, which is intended to trap [unintended return type errors](http://eslint.org/docs/rules/consistent-return), now that we use Flow. 

> If any code paths in a function return a value explicitly but some code path do not return a value explicitly, it might be a typing mistake, especially in a large function

Perhaps we can decommission this rule in the `javascripts` folder.